jdaviz/configs/imviz/plugins/compass/compass.py

```diff
@@ -46,6 +49,18 @@ def _on_viewer_data_changed(self, msg=None):
             viewer = self.viewer.selected_obj
             viewer.on_limits_change()  # Force redraw
 
+    def _on_canvas_rotation_changed(self, msg=None):
+        viewer_id = msg.viewer_id
+        if viewer_id != self.viewer_selected:
+            return
+
+        self._set_compass_rotation()
+
+    def _set_compass_rotation(self):
+        viewer_item = self.app._viewer_item_by_id(self.viewer.selected_id)
+        self.canvas_angle = viewer_item.get('canvas_angle', 0)  # noqa

```

Comment:
Nitpick: Not sure why `# noqa` is needed here.

```suggestion
        self.canvas_angle = viewer_item.get('canvas_angle', 0)
```